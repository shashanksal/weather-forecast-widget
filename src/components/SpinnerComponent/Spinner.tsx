import React from "react";
import { usePromiseTracker } from "react-promise-tracker"; //Library for tracking Fetch, Axios reqeusts
import Loader from "react-loader-spinner";

const spinneStyles = {
    width: '100%',
    height: '100%',
    position: "relative",
    bottom: "80vh",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
} as React.CSSProperties;

export const Spinner: any = () => {
    const { promiseInProgress } = usePromiseTracker();
    return (
        promiseInProgress && (
            <div style={spinneStyles}>
                <Loader type="Grid" color="#009688" height={200} width={200} />
            </div>
        )
    );
};
