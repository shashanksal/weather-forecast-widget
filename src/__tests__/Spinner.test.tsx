import React from "react";
import { render } from "@testing-library/react";

import { Spinner } from "../components/SpinnerComponent";

describe("test category", () => {
	it("Spinner matches snapshot", () => {
		const { asFragment } = render(<Spinner />);

		expect(asFragment()).toMatchSnapshot();
	});
});
