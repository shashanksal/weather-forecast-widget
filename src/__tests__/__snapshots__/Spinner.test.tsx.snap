// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test category Spinner matches snapshot 1`] = `<DocumentFragment />`;
